src/drivers/uavcan/node_info.cpp

```diff
@@ -0,0 +1,335 @@
+		return;
+	}
+
+	_last_device_info_publish = now;
+	publishSingleDeviceInformation(device_info);
+}
+
+void NodeInfoPublisher::publishDeviceInformationPeriodic()
+{
+	// Using round-robin approach to publish one device info per timer event
+	if (_device_informations_size == 0) {
+		return;
+	}
+
+	size_t devices_checked = 0;
+
+	while (devices_checked < _device_informations_size) {
+		if (_next_device_to_publish >= _device_informations_size) {
+			_next_device_to_publish = 0;
+		}
+
+		const auto &device_info = _device_informations[_next_device_to_publish];
+
+		if (device_info.has_node_info && device_info.has_capability) {
+			publishSingleDeviceInformation(device_info);
+			_next_device_to_publish++;
+			return;
+		}
+
+		_next_device_to_publish++;
+		devices_checked++;
+	}
+
+	PX4_DEBUG("No devices ready for periodic publishing");
+}
+
+void NodeInfoPublisher::publishSingleDeviceInformation(const DeviceInformation &device_info)
+{
+	const uint64_t now = hrt_absolute_time();
+
+	device_information_s msg{};
+	msg.timestamp = now;
+	msg.device_type = static_cast<uint8_t>(device_info.capability);
+	msg.device_id = device_info.device_id;
+
+	// Copy pre-populated fields directly from the struct
+	strlcpy(msg.model_name, device_info.model_name, sizeof(msg.model_name));
+	strlcpy(msg.vendor_name, device_info.vendor_name, sizeof(msg.vendor_name));
+	strlcpy(msg.firmware_version, device_info.firmware_version, sizeof(msg.firmware_version));
+	strlcpy(msg.hardware_version, device_info.hardware_version, sizeof(msg.hardware_version));
+	strlcpy(msg.serial_number, device_info.serial_number, sizeof(msg.serial_number));
+
+	_device_info_pub.publish(msg);
+
+	PX4_DEBUG("Published device info for node %d, device_id %lu, type %d",
+		  device_info.node_id, static_cast<unsigned long>(device_info.device_id),
+		  static_cast<int>(device_info.capability));
+}
+
+void NodeInfoPublisher::populateDeviceInfoFields(DeviceInformation &device_info, const NodeInfo &info)
+{
+	device_info.has_node_info = true;
+
+	// Parse the node name to extract vendor and model information
+	parseNodeName(info.name, device_info);
+
+	snprintf(device_info.firmware_version, sizeof(device_info.firmware_version),
+		 "%d.%d.%lu", info.sw_major, info.sw_minor, static_cast<unsigned long>(info.vcs_commit));
+	snprintf(device_info.hardware_version, sizeof(device_info.hardware_version),
+		 "%d.%d", info.hw_major, info.hw_minor);
+	snprintf(device_info.serial_number, sizeof(device_info.serial_number),
+		 "%02x%02x%02x%02x%02x%02x%02x%02x",
+		 info.unique_id[0], info.unique_id[1], info.unique_id[2], info.unique_id[3],
+		 info.unique_id[4], info.unique_id[5], info.unique_id[6], info.unique_id[7]);
+}
+
+void NodeInfoPublisher::parseNodeName(const char *name, DeviceInformation &device_info)
+{
+	if (!name || strlen(name) == 0) {
+		strlcpy(device_info.vendor_name, "-1", sizeof(device_info.vendor_name));
+		strlcpy(device_info.model_name, "-1", sizeof(device_info.model_name));
```

Comment:
Why not keep it empty?
```suggestion
		strlcpy(device_info.vendor_name, "", sizeof(device_info.vendor_name));
		strlcpy(device_info.model_name, "", sizeof(device_info.model_name));
```